### ISSUE TITLE
Variant-generator

### ISSUE THREAD

**Issue by Lingkar** [stanford-mast/INFaaS#5]
Hi, I am trying to replicate some of your work. 
But I have some questions regarding the variant generator.
Is it located in the source code, as I have not been able to find it?

In the case, it is not available I have the following questions regarding the generation of models in your work.
For GPU variants you used TensorRT.
For Neuron instances, you used the Neuron compiler to create different variants.
Did you generate model variants for CPU instances, and if so how? 

It is stated that for the ResNet50 model INFaaS generates 50 variants. 
How exactly did the generator end up with around 50 models?
If I am correct, you got 7 different variants based on changing batch size for each hardware platform.
For TensorRT you changed the datatype to FP16 if I am correct?
It would be very helpful to get a bit more of a detailed perspective on the generating of model variants, thank you very much.

**Comment by faromero** [comment]
Hello, thanks for trying out INFaaS! Here's some answers to your questions:
* At this time the variant generation is a manual operation; we also use open-source, pre-trained variants (e.g., we used AWS's Neuron tutorial to generate ResNet50 variants for Inferentia).
* For CPU instances, we did not use any compilers, but we did consider different numbers of cores (i.e., using 1CPU vs 4CPU is two different variants). There are optimization frameworks for CPU that can be used, but we have not considered them at the moment.
* Yes, you are correct. We generated our variants by varying: different batch sizes, using TensorRT with FP16, using Neuron with different numbers of Inferentia cores, using different numbers of CPU cores (see bulletpoint above), and using different frameworks (PyTorch, Caffe, TensorFlow).
